packages/teleport/src/Databases/AddDatabase/AddDatabase.tsx

```diff
@@ -22,21 +22,52 @@ import Dialog, {
   DialogFooter,
 } from 'design/Dialog';
 import { Text, Box, ButtonSecondary, Link } from 'design';
-import Select, { Option } from 'shared/components/Select';
+import { Danger } from 'design/Alert';
+import { Attempt } from 'shared/hooks/useAttemptNext';
+import Select, { DarkStyledSelect, Option } from 'shared/components/Select';
 import * as links from 'teleport/services/links';
 import { AuthType } from 'teleport/services/user';
 import { DbType, DbProtocol } from 'teleport/services/databases';
+import { JoinToken } from 'teleport/services/jointoken';
+import useTeleport from 'teleport/useTeleport';
+import useDatabases from '../useDatabases';
+import useAddDatabase from './useAddDatabase';
 import {
   formatDatabaseInfo,
   DatabaseInfo,
 } from 'teleport/services/databases/makeDatabase';
 import TextSelectCopy from 'teleport/components/TextSelectCopy';
 
-export default function AddDatabase({
+export default function Container() {
+  const ctx = useTeleport();
+  const {
+    username,
+    version,
+    authType,
+    canCreate,
+    hideAddDialog,
+  } = useDatabases(ctx);
+  const { joinToken, attempt } = useAddDatabase(ctx);
```

Comment:
this `useDatabases(ctx)` will kick in a new initialization flow for the `Database` feature, instead pass the upper level function via props.

I am not sure if it even works (it wont close the dialog). Have you tried running it against real Teleport instance and not the Storybook?

Reply:
I did a round of cleanups and created a branch, here is my commit with cleanups https://github.com/gravitational/webapps/commit/270f9295af880e0f150a48165459ddd72a6c8969 .

Reply:
Updated the link https://github.com/gravitational/webapps/commit/a39d798191d0fd1b6dc53d265b07fcb67497fcd6